### ISSUE TITLE
Plugin registered only for a session pollutes the global plugin state

### ISSUE THREAD

**Issue by presidento** [nushell/nushell#8581]
### Describe the bug

The help says:

```
❯ help register
Register a plugin.

[...]

  Register `nu_plugin_query` plugin from `nu -c`(plugin will be available in that nu session only)
  > let plugin = ((which nu).path.0 | path dirname | path join 'nu_plugin_query'); nu -c $'register ($plugin); version'
```

When I run register with `nu -c`, the plugin still will be written in `plugins.nu`, so next time I start Nushell it will be available globally.

## ✔ Check that there is no registered `inc` command

```
❯ "1.2.0" | inc --major
Error: nu::shell::external_command

  × External command failed
   ╭─[entry #28:1:1]
 1 │ "1.2.0" | inc --major
   ·           ─┬─
   ·            ╰── executable was not found
   ╰────
  help: program not found
```

## ✔ Register and use it only for a session

```
❯ nu -c 'register "c:/Program Files/nu/bin/nu_plugin_inc.exe"; "1.2.0" | inc --major'
2.0.0
```

## ✔ Check that it is not available in the current session

```
❯ "1.2.0" | inc --major
  × External command failed
   ╭─[entry #30:1:1]
 1 │ "1.2.0" | inc --major
   ·           ─┬─
   ·            ╰── executable was not found
   ╰────
  help: program not found
```

## ❌ Start a new shell, and there it is

```
❯ nu -c ' "1.2.0" | inc --major '
2.0.0
```

### Expected behavior

It would be good to register a plugin only for a session, which requires the plugin info not to be persisted in the `plugins.nu` file.

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.77.1                                   |
| branch             |                                          |
| commit_hash        | 0120e4040d09f3bdffb945fa5fe450b72df7837a |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| build_time         | 2023-03-17 05:15:37 +00:00               |
| build_rust_channel | release                                  |
| features           | default, zip                             |
| installed_plugins  | py-multiplier, py-noop                   |

**Comment by fdncred** [comment]
Right, this is by design. Plugins are not meant to be used for one session only. If you want to do that, you'll need to `rm $nu.plugin-path` when you're done with the plugin. 

The point here is that registering plugins is a tool used to write the $nu.plugin-path file so that nushell knows what plugins are available to the system on the next launch. Or, if you do it like `register /path/to/nu_plugin_something`, it's available in this current session too (as opposed to shelling out with nu -c)

This part of the help `(plugin will be available in that nu session only)` is wrong. We shouldn't have allowed that in to begin with, so I can see where the confusion is.
